Fix the build on linux

martin-g opened the pull request:
Fixes #9 

1. Replace SINT_MAX (from R.h ?!) with INT8_MAX (from stdint.h)
    
2. Pass the result of Rhtslib::pkgconfig("PKG_LIBS") to `tr -d "'"`
    
    because it returns it wrapped in single quotes and the linker (ld)
    cannot file the file:
    
    `/usr/bin/ld: cannot find '/home/biocbuild/bbs-3.17-bioc/R/library/Rhtslib/usrlib/libhts.a': No such file or directory`
    
 3. Fix a minor typo

martin-g committed:
Fix a minor typo

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Pass the result of Rhtslib::pkgconfig("PKG_LIBS") to `tr -d "'"`

because it returns it wrapped in single quotes and the linker (ld)
cannot file the file:

/usr/bin/ld: cannot find '/home/biocbuild/bbs-3.17-bioc/R/library/Rhtslib/usrlib/libhts.a': No such file or directory

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Replace SINT_MAX (from R.h ?!) with INT8_MAX (from stdint.h)

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

grimbough commented:
Just saw this in the Slack channel and have a few thoughts.

Do you really want to use `INT8_MAX` as the replacement here?  Wouldn't `INT_MAX` be the more direct replacement, since that's all `SINT_MAX` was defined as anyway (see https://github.com/wch/r-source/commit/68804440c3674b054d23b701ceb46753bb0abe2b#)

Regarding the removal of the single quotes, what happens if your library path has a space in it?  I've definitely encountered linking problems when that's the case.  

It also looks like the quotes come from Rhtslib.  If they're really the problem, might it be better to fix the problem at source so it doesn't affect other packages too?  In Rhdf5lib it looks like I choose to use double quotes - perhaps that's more robust e.g. https://github.com/grimbough/Rhdf5lib/blob/7b411e1478e43ebfe0ca201c5b8c12640bd9e18d/R/zzz.R#L62

martin-g committed:
Improvements after code review by @grimbough

* Use INT_MAX instead of INT8_MAX
* Replace single quotes with double quotes when wrapping PKG_LIBS.
  Related to https://github.com/Bioconductor/Rhtslib/commit/f1c2d57e8a49ad9ab472a9341aeeaa80ac81feae#diff-fbc888a8d95ca11d7f17e5eaa27a08a3d3cb183803ae509deea6aa51f61183fbR14

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
Thank you for the review, @grimbough !
I've made the change from `INT8_MAX` to `INT_MAX`!
I also changed the removal of `'` to replacing it with `"`.

@hpages Could you also please take a look ?
You introduced the the single quotes with https://github.com/Bioconductor/Rhtslib/commit/f1c2d57e8a49ad9ab472a9341aeeaa80ac81feae#diff-fbc888a8d95ca11d7f17e5eaa27a08a3d3cb183803ae509deea6aa51f61183fbR14. I am not sure how but it worked for you for 4 years.

hpages commented:
Hi @martin-g

> I am not sure how but [the single quotes] worked for you for 4 years.

Also worked and still works for all packages that are rev deps (via the `LinkingTo` field) of **Rhtslib**, e.g. **bamsignals**, **csaw**, **deepSNV**, etc...: https://bioconductor.org/checkResults/3.17/bioc-LATEST/Rhtslib/

Furthermore, the single quotes don't seem to cause problems for these packages on your Linux ARM64 machine kunpeng1:
- https://yikun.github.io/bioconductor-0210/report/bamsignals/kunpeng1-install.html
- https://yikun.github.io/bioconductor-0210/report/csaw/kunpeng1-install.html
- etc...

so it's really not clear to me why they are a problem for **qrqc**.

Can you please try using double quotes instead of single quotes on your Linux ARM64 machine? Should be just a matter of replacing

    usrlib_path <- sprintf("'%s'", file.path(usrlib_dir, "libhts.a"))

with

    usrlib_path <- sprintf("\"%s\"", file.path(usrlib_dir, "libhts.a"))

in `Rhtslib/R/zzz.R`, re-installing **Rhtslib**, and try to install **qrqc** again.

If that works then that would be a better way to address the problem. But it would be good to actually observe the single quote problem [on kunpeng1](https://yikun.github.io/bioconductor-0210/report/qrqc/kunpeng1-install.html) before trying to address it.

The `SINT_MAX` issue is caused by a recent change in R devel and it breaks **qrqc** on all platforms:
https://bioconductor.org/checkResults/3.17/bioc-LATEST/qrqc/ I agree with @grimbough that this should be replaced with `INT_MAX` because that's how `SINT_MAX` was defined.

@vsbuffalo Can you please make that change? Thanks!

martin-g commented:
I've found the problem!
https://github.com/vsbuffalo/qrqc/blob/master/src/Makevars#L1 -here the `echo` wraps the value in single quotes!
https://github.com/lamortenera/bamsignals/blob/master/src/Makevars#L5 - does not use `echo` and single quotes.
I'll update the PR tomorrow morning.
The `INT_MAX` change is already part of this PR.

hpages commented:
Wait a minute, this Makevars file is 4 year old! The Makevars file in the **qrqc** package at git.bioconductor.org (`git clone git@git.bioconductor.org:packages/qrqc.git`) is more recent and does the right thing:
```
## This file uses GNU make syntax $(shell ...) so we need to
## have "SystemRequirements: GNU make" in the DESCRIPTION file.
## See Rhtslib's vignette for details.

RHTSLIB_LIBS=$(shell "${R_HOME}/bin${R_ARCH_BIN}/Rscript" -e \
    'Rhtslib::pkgconfig("PKG_LIBS")')
RHTSLIB_CPPFLAGS=$(shell "${R_HOME}/bin${R_ARCH_BIN}/Rscript" -e \
    'Rhtslib::pkgconfig("PKG_CPPFLAGS")')

PKG_LIBS=$(RHTSLIB_LIBS)
PKG_CPPFLAGS=$(RHTSLIB_CPPFLAGS)
```
@vsbuffalo Can you please resync the GitHub version of the package with the version at git.bioconductor.org?

@martin-g Note that Bioconductor builds, checks, and distributes packages hosted on our git server at git.bioconductor.org. It turns out that **qrqc** problems have nothing to do with Linux ARM64 so please remove it from your list (I don't quite get how it ended up on your list in the first place because I don't see it mentioned here https://github.com/Bioconductor/BBS/pull/255). Thanks!

martin-g commented:
Closing the issue (#9) and this PR!

We track the Linux ARM64 issues at https://github.com/Yikun/BBS/issues/ (qrqc is https://github.com/Yikun/BBS/issues/17).
We know that git.bioconductor.org is used by BBS, but we use the Github mirrors because we can help the maintainers by sending Pull Requests.

hpages commented:
I see but since it fails the same way on Linux x86_64 you can also remove the issue from Yikun/BBS list.
